src/ctap/mod.rs

```diff
@@ -838,13 +849,19 @@ impl CtapState {
                     return Err(Ctap2StatusCode::CTAP2_ERR_PUAT_REQUIRED);
                 }
                 // Corresponds to makeCredUvNotRqd set to true.
-                if options.rk && storage::pin_hash(env)?.is_some() {
+                if options.rk
+                    && slot_id.is_some()
+                    && storage::pin_hash(env, slot_id.unwrap())?.is_some()
+                {
                     return Err(Ctap2StatusCode::CTAP2_ERR_PUAT_REQUIRED);
                 }
                 0x00
             }
         };
         flags |= UP_FLAG | AT_FLAG;
+        // By this point, we should have already returned the suitable errors in all situations that
+        // slot_id is none.
+        let slot_id = slot_id.ok_or(Ctap2StatusCode::CTAP2_ERR_VENDOR_INTERNAL_ERROR)?;
```

Comment:
What about the case where the feature is not enabled? Not covered by this PR?

Reply:
This PR actually only covers cases when the feature isn't enabled.
When it isn't enabled, slot_id will be `Some(0)`, as requests always meant to access the PIN on slot 0.

Reply:
Got it. Was hard to infer that, given the incomplete state of the PR.

Reply:
I provided the rationale in the commit message, do you think that's enough? Or should I add more comments in the code?

Reply:
Oh, should have read that more carefully, my bad.